csrc/ir/nodes.cpp

```diff
@@ -3287,16 +3287,36 @@ bool TensorDomain::sameAs(
   return true;
 }
 
-std::string TensorDomain::toString(int indent_size) const {
+std::string TensorDomain::toString(const int indent_size, const bool leaf_only)
+    const {
   std::stringstream ss;
   if (nDims() == 0) {
-    ss << "[ 0 ]";
```

Comment:
I think `[0]` is a bug. `nDims()==0` means a scalar tensor but `[0]` means a vector of size 0. Let me know otherwise.

Reply:
Not sure what is a bug. A vector of size 0 here means a TensorDomain has 0 IterDomain, which means a scalar tensor, right?

Reply:
Oh, you meant `[0]` means a vector having a value printed as `0`, which indicates it has one axis of extent 0? That indeed seems confusing. Agreed with the change.